app/controllers/application_controller.rb

```diff
@@ -1,5 +1,36 @@
 class ApplicationController < ActionController::Base
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  def set_flash(msg)
+    flash[:notice] = msg
+  end
+
+  def current_user
+    session[:user_id] ? User.find(session[:user_id]) : nil
+  end
+
+  def session_in!(user)
+    session[:user_id] = user.id
+  end
+
+  def session_out!
+    session[:user_id] = nil
```

Comment:
These two are another case where you probably would be better off just inlining the content at the call site. I think the only place these are legitimately called is sessions_controller so you could just manipulate the session hash there. Single-line methods are only really valuable if they reveal an intention that would otherwise be less clear.